hack/build-client.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+BUILD_WORKDIR="${PWD}"
+
+function clean() {
+  rm ${BUILD_WORKDIR}/.sha256sum
+
+  rm -rf ${BUILD_WORKDIR}/pkg/client
+  mkdir ${BUILD_WORKDIR}/pkg/client
+
+  rm -rf ${BUILD_WORKDIR}/python/client
+  mkdir -p ${BUILD_WORKDIR}/python/client
+}
+
+function checksum() {
+  sha256sum swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/"$1"/redhatopenshift.json >> .sha256sum
+}
+
+function generate_golang() {
+  sudo docker run \
+		--rm \
+		-v ${BUILD_WORKDIR}/pkg/client:/github.com/Azure/ARO-RP/pkg/client:z \
+		-v ${BUILD_WORKDIR}/swagger:/swagger:z \
+		azuresdk/autorest \
+		--go \
+		--license-header=MICROSOFT_APACHE_NO_VERSION \
+		--namespace=redhatopenshift \
+		--input-file=/swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/"$1"/redhatopenshift.json \
+		--output-folder=/github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/"$1"/redhatopenshift
+
+  sudo chown -R $(id -un):$(id -gn) ${BUILD_WORKDIR}/pkg/client
+  sed -i -e 's|azure/aro-rp|Azure/ARO-RP|g' ${BUILD_WORKDIR}/pkg/client/services/redhatopenshift/mgmt/"$1"/redhatopenshift/models.go ${BUILD_WORKDIR}/pkg/client/services/redhatopenshift/mgmt/"$1"/redhatopenshift/redhatopenshiftapi/interfaces.go
+  go run ${BUILD_WORKDIR}/vendor/golang.org/x/tools/cmd/goimports -w -local=github.com/Azure/ARO-RP ${BUILD_WORKDIR}/pkg/client
```

Comment:
It looks like we can run it only once after generating clients for all version.

Reply:
Actually no, please see `$1` on line 32. It is run for each package.
Also, the `chown` has to be run before every generation. I also thought this can be generalized, but it turned out there are dependencies for each run.